2.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from openai import OpenAI
+
+client = OpenAI(base_url="http://localhost:8000/v1", api_key="fake")
+
+completion = client.chat.completions.create(
+    model="meta-llama/Llama-3.2-3B-Instruct",
+    messages=[
+        {"role": "user", "content": "Write me a haiku about coding"},
+    ],
+    response_format={
+        "type": "json_schema",
+        "json_schema": {
+            "name": "name_of_response_format",
+            "schema": {
+                "type": "object",
+                "properties": {
+                    "completion_message": {
+                        "type": "string",
+                    }
+                },
+                "required": ["completion_message"],
+            },
+            "strict": True,
+        },
+    },
+)
+
+print(completion.choices[0].message.content)
```

Comment:
Outputs:
```
Lines of code descend
Algorithmic symphony
Logic's gentle song
```

vLLM seems to ignore the field:

```
INFO 11-26 02:06:15 logger.py:37] Received request chat-2cf699b3a6dd4580b97749c3c39d9b2b: prompt: '<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\nCutting Knowledge Date: December 2023\nToday Date: 26 Nov 2024\n\n<|eot_id|><|start_header_id|>user<|end_header_id|>\n\nWrite me a haiku about coding<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n',
params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.7, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=[], stop_token_ids=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=131030, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None), guided_decoding=GuidedDecodingParams(json=None, regex=None, choice=None, grammar=None, json_object=None, backend=None, whitespace_pattern=None),
prompt_token_ids: [128000, 128006, 9125, 128007, 271, 38766, 1303, 33025, 2696, 25, 6790, 220, 2366, 18, 198, 15724, 2696, 25, 220, 1627, 4723, 220, 2366, 19, 271, 128009, 128006, 882, 128007, 271, 8144, 757, 264, 6520, 39342, 922, 11058, 128009, 128006, 78191, 128007, 271],
lora_request: None,
prompt_adapter_request: None.
```

Specifically:
```
guided_decoding=GuidedDecodingParams(json=None, regex=None, choice=None, grammar=None, json_object=None, backend=None, whitespace_pattern=None)
```

`GuidedDecodingParams` is not populated.

Reply:
Note: I also tried  

```
    extra_body={
        "response_format": ...
    },
```

also doesn't work: https://github.com/aidando73/llama-stack/pull/1/files#r1857894484